diff --git a/node_modules/@edx/paragon/dist/DataTable/selection/BaseSelectionStatus.js b/node_modules/@edx/paragon/dist/DataTable/selection/BaseSelectionStatus.js
index 00c4c4b..426ca4e 100644
--- a/node_modules/@edx/paragon/dist/DataTable/selection/BaseSelectionStatus.js
+++ b/node_modules/@edx/paragon/dist/DataTable/selection/BaseSelectionStatus.js
@@ -20,11 +20,12 @@ function BaseSelectionStatus(_ref) {
     itemCount,
     filteredRows,
     isPaginated,
-    state
+    state,
+    controlledTableSelections,
   } = useContext(DataTableContext);
   const hasAppliedFilters = state?.filters?.length > 0;
-  const isAllRowsSelected = numSelectedRows === itemCount;
-  const filteredItems = filteredRows?.length || itemCount;
+  const isAllRowsSelected = controlledTableSelections?.isEntireTableSelected || numSelectedRows === itemCount;
+  const selectAllItemCount = itemCount;
   const intlAllSelectedText = allSelectedText || /*#__PURE__*/React.createElement(FormattedMessage, {
     id: "pgn.DataTable.BaseSelectionStatus.allSelectedText",
     defaultMessage: "All {numSelectedRows} selected",
@@ -62,7 +63,7 @@ function BaseSelectionStatus(_ref) {
     defaultMessage: "Select all {itemCount}",
     description: "A label for select all button.",
     values: {
-      itemCount: filteredItems
+      itemCount: selectAllItemCount
     }
   })), numSelectedRows > 0 && /*#__PURE__*/React.createElement(Button, {
     className: CLEAR_SELECTION_TEST_ID,
diff --git a/node_modules/@edx/paragon/icons/es5/ManageAccounts.js b/node_modules/@edx/paragon/icons/es5/ManageAccounts.js
index 2246911..8940d48 100644
--- a/node_modules/@edx/paragon/icons/es5/ManageAccounts.js
+++ b/node_modules/@edx/paragon/icons/es5/ManageAccounts.js
@@ -12,7 +12,8 @@ const SvgManageAccounts = props => /*#__PURE__*/React.createElement("svg", _exte
 }), /*#__PURE__*/React.createElement("circle", {
   cx: 10,
   cy: 8,
-  r: 4
+  r: 4,
+  fill: "currentColor"
 }), /*#__PURE__*/React.createElement("path", {
   d: "M20.75 16c0-.22-.03-.42-.06-.63l1.14-1.01-1-1.73-1.45.49c-.32-.27-.68-.48-1.08-.63L18 11h-2l-.3 1.49c-.4.15-.76.36-1.08.63l-1.45-.49-1 1.73 1.14 1.01c-.03.21-.06.41-.06.63s.03.42.06.63l-1.14 1.01 1 1.73 1.45-.49c.32.27.68.48 1.08.63L16 21h2l.3-1.49c.4-.15.76-.36 1.08-.63l1.45.49 1-1.73-1.14-1.01c.03-.21.06-.41.06-.63zM17 18c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z",
   fill: "currentColor"
diff --git a/node_modules/@edx/paragon/icons/jsx/ManageAccounts.jsx b/node_modules/@edx/paragon/icons/jsx/ManageAccounts.jsx
index 5491289..c1f8773 100644
--- a/node_modules/@edx/paragon/icons/jsx/ManageAccounts.jsx
+++ b/node_modules/@edx/paragon/icons/jsx/ManageAccounts.jsx
@@ -12,7 +12,7 @@ const SvgManageAccounts = (props) => (
       d="M10.67 13.02c-.22-.01-.44-.02-.67-.02-2.42 0-4.68.67-6.61 1.82-.88.52-1.39 1.5-1.39 2.53V20h9.26a6.963 6.963 0 0 1-.59-6.98z"
       fill="currentColor"
     />
-    <circle cx={10} cy={8} r={4} />
+    <circle cx={10} cy={8} r={4} fill="currentColor"/>
     <path
       d="M20.75 16c0-.22-.03-.42-.06-.63l1.14-1.01-1-1.73-1.45.49c-.32-.27-.68-.48-1.08-.63L18 11h-2l-.3 1.49c-.4.15-.76.36-1.08.63l-1.45-.49-1 1.73 1.14 1.01c-.03.21-.06.41-.06.63s.03.42.06.63l-1.14 1.01 1 1.73 1.45-.49c.32.27.68.48 1.08.63L16 21h2l.3-1.49c.4-.15.76-.36 1.08-.63l1.45.49 1-1.73-1.14-1.01c.03-.21.06-.41.06-.63zM17 18c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
       fill="currentColor"
